front/lib/resources/remote_mcp_server_tool_metadata_resource.ts

```diff
@@ -0,0 +1,147 @@
+import { assert } from "console";
+import type {
+  Attributes,
+  CreationAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+
+import type { MCPToolPermissionType } from "@app/lib/actions/mcp_metadata";
+import type { Authenticator } from "@app/lib/auth";
+import { RemoteMCPServerToolMetadata } from "@app/lib/models/assistant/actions/remote_mcp_server_tool_metadata";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { SpaceResource } from "@app/lib/resources/space_resource";
+import type { ResourceFindOptions } from "@app/lib/resources/types";
+import type { Result } from "@app/types";
+import { Err, Ok } from "@app/types";
+
+export class RemoteMCPServerToolMetadataResource extends BaseResource<RemoteMCPServerToolMetadata> {
+  static model: ModelStatic<RemoteMCPServerToolMetadata> =
+    RemoteMCPServerToolMetadata;
+
+  constructor(
+    model: typeof RemoteMCPServerToolMetadata,
+    blob: Attributes<RemoteMCPServerToolMetadata>
+  ) {
+    super(RemoteMCPServerToolMetadata, blob);
+  }
+
+  // Creation
+
+  static async makeNew(
+    auth: Authenticator,
+    blob: CreationAttributes<RemoteMCPServerToolMetadata>,
+    transaction?: Transaction
+  ) {
+    const systemSpace = await SpaceResource.fetchWorkspaceSystemSpace(auth);
+    assert(
+      systemSpace.canWrite(auth),
+      "The user is not authorized to create a tool metadata"
+    );
+
+    const toolMetadata = await this.model.create(blob, { transaction });
+
+    return new this(RemoteMCPServerToolMetadata, toolMetadata.get());
+  }
+
+  // Fetch
+
+  private static async baseFetch(
+    auth: Authenticator,
+    options: ResourceFindOptions<RemoteMCPServerToolMetadata>
+  ) {
+    const { where, ...opts } = options;
+
+    const toolMetadata = await RemoteMCPServerToolMetadata.findAll({
+      where: {
+        workspaceId: auth.getNonNullableWorkspace().id,
+        ...where,
```

Comment:
```suggestion
        ...where,
        workspaceId: auth.getNonNullableWorkspace().id,
```

Otherwise, we can override the `workspaceId` which we definitely don't want here.,